stdlib/queue.gr

```diff
@@ -1,46 +1,115 @@
+/**
+ * @module Queue: An immutable queue implementation. A queue is a FIFO (first-in-first-out) data structure where new values are added to the end and retrieved or removed from the beginning.
+ * @example import Queue from "queue"
+ * @since v0.2.0
+ */
 import List from "list"
 
 record Queue<a> { forwards: List<a>, backwards: List<a> }
 
+/**
+ * @section Values: Functions for working with queues.
+ */
+
+/**
+ * Creates an empty queue.
+ * 
+ * @returns An empty queue
+ * @since v0.2.0
+ */
 export let make = () => {
-    { forwards: [], backwards: [] }
+  { forwards: [], backwards: [] }
 }
 
-export let isEmpty = (queue) => {
-    match (queue) {
-        { forwards: [], backwards: [] } => true,
-        _ => false
-    }
+/**
+ * Checks if the given queue contains any values.
+ * 
+ * @param queue: The queue to check
+ * @returns `true` if the given queue is empty or `false` otherwise
+ *
+ * @since v0.2.0
+ */
+export let isEmpty = queue => {
+  match (queue) {
+    { forwards: [], backwards: [] } => true,
+    _ => false,
+  }
 }
 
-export let peek = (queue) => {
-    match (queue) {
-        { forwards: [], backwards: [] } => None,
-        { forwards, backwards } => List.head(forwards)
-    }
+/**
+ * Returns the value at the beginning of the queue, without removing the value from the queue.
```

Comment:
Because we have the docs above that explain how queues work, I think we can get away with just saying "the next value", but curious to hear what others think.

For this one, it feels like it implies that there's a function that will return the value while also removing it—I think a slight rewording fixes that for me:
```suggestion
 * Returns the value at the beginning of the queue. It is not removed from the queue.
```

Reply:
Returns the value at the beginning of the queue without removing it.